app/scripts/shell/dimSearchFilter.directive.js

```diff
@@ -685,6 +709,116 @@
       },
       transferable: function(predicate, item) {
         return !item.notransfer;
+      },
+      rof: function(predicate, item) {
+        return filterByStats.do(predicate, item, 'rof');
+      },
+      impact: function(predicate, item) {
+        return filterByStats.do(predicate, item, 'impact');
+      },
+      range: function(predicate, item) {
+        return filterByStats.do(predicate, item, 'range');
+      },
+      stability: function(predicate, item) {
+        return filterByStats.do(predicate, item, 'stability');
+      },
+      reload: function(predicate, item) {
+        return filterByStats.do(predicate, item, 'reload');
+      },
+      magazine: function(predicate, item) {
+        return filterByStats.do(predicate, item, 'magazine');
+      },
+      aimassist: function(predicate, item) {
+        return filterByStats.do(predicate, item, 'aa');
+      },
+      equipspeed: function(predicate, item) {
+        return filterByStats.do(predicate, item, 'equipspeed');
+      }
+    };
+
+    // This refactored method filters items by stats
+    //   * statType = [aa|impact|range|stability|rof|reload|magazine|equipspeed]
+    var filterByStats = {
+      do: function(predicate, item, statType) {
+        if (predicate.length === 0 || item.stats === undefined) {
+          return false;
+        }
+
+        var operands = ['<=', '>=', '=', '>', '<'];
+        var operand = 'none';
+        var result = false;
+        var statHash = 0;
+
+        operands.forEach(function(element) {
+          if (predicate.substring(0, element.length) === element) {
+            operand = element;
+            predicate = predicate.substring(element.length);
+            return false;
+          } else {
+            return true;
+          }
+        }, this);
+
+        switch (statType) {
+        case 'impact': // Impact
+          statHash = 4043523819;
+          break;
+        case 'range': // Range
+          statHash = 1240592695;
+          break;
+        case 'stability': // Stability
+          statHash = 155624089;
+          break;
+        case 'rof': // Rate of fire
+          statHash = 4284893193;
+          break;
+        case 'reload': // Reload
+          statHash = 4188031367;
+          break;
+        case 'magazine': // Magazine
+          statHash = 387123106;
+          break;
+        case 'aa': // Aim Assist
+          statHash = 1345609583;
+          break;
+        case 'equipspeed': // Equip Speed
+          statHash = 943549884;
+          break;
+        }
+
+        item.stats.forEach(function(stat) {
+          if (stat.statHash === statHash) {
+            item.stats.value = stat.value;
```

Comment:
I'm a bit confused - are you creating a new `value` property on `stats`? Can we avoid that?